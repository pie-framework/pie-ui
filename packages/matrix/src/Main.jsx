import React from 'react';
import PropTypes from 'prop-types';
import {PreviewLayout} from '@pie-lib/render-ui';
import Matrix from './Matrix';

class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onSessionChange: PropTypes.func
  };

  static defaultProps = {
    model: {},
    session: {},
  };

  render() {
    const {model, onSessionChange, session} = this.props;
    return (
      <PreviewLayout>
        <Matrix
          {...model}
          session={session}
          onSessionChange={onSessionChange}
        />
      </PreviewLayout>
    );
  }
}

export default Main;
