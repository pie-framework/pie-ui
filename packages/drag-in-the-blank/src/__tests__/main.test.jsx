import { shallow } from 'enzyme';
import React from 'react';

import Main from '../main';

jest.mock('@pie-lib/render-ui', () => ({
  hasText: jest.fn().mockReturnValue('true'),
  color: {
    text: jest.fn().mockReturnValue('black'),
    background: jest.fn().mockReturnValue('grey'),
  }
}));

describe('Main', () => {
  let wrapper;
  let onChange = jest.fn();

  beforeAll(() => {
    wrapper = (extra) => {
      const props = {
        model: {
          prompt: 'Prompt',
          mode: 'gather',
          rationale: 'Rationale',
          teacherInstructions: 'Teacher Instructions',
          rationaleEnabled: true,
          promptEnabled: true,
          teacherInstructionsEnabled: true,
          studentInstructionsEnabled: true,
          id: '1',
          element: 'drag-in-the-blank',
          markup: '{{0}} + {{1}} = 15',
          disabled: false,
          shuffle: true,
          choices: [{ value: '<div>9</div>', id: '1' }, { value: '<div>6</div>', id: '0' }],
          choicesPosition: 'below',
          correctResponse: { 0: '0', 1: '1' },
          duplicates: true,
          alternateResponses: [['1'], ['0']],
          feedback: {},
          ...extra
        },
        value: { 0: '1', 1: '0' },
        onChange
      };

      return shallow(<Main {...props}/>);
    };
  });

  describe('render', () => {
    it('should render in gather mode', () => {
      expect(wrapper()).toMatchSnapshot();
    });

    it('should render in view mode', () => {
      expect(wrapper({ mode: 'view' })).toMatchSnapshot();
    });

    it('should render in evaluate mode', () => {
      expect(wrapper({ mode: 'evaluate' })).toMatchSnapshot();
    });

    it('should render without teacher instructions', () => {
      expect(wrapper({ teacherInstructions: null })).toMatchSnapshot();
    });

    it('should render without rationale', () => {
      expect(wrapper({ rationale: null })).toMatchSnapshot();
    });

    it('should render without prompt', () => {
      expect(wrapper({ prompt: null })).toMatchSnapshot();
    });
  });
});
