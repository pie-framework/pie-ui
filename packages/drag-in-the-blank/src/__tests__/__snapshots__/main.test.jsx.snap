// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main render should render in evaluate mode 1`] = `
<Main
  classes={
    Object {
      "mainContainer": "Main-mainContainer-1",
    }
  }
  model={
    Object {
      "alternateResponses": Array [
        Array [
          "1",
        ],
        Array [
          "0",
        ],
      ],
      "choices": Array [
        Object {
          "id": "1",
          "value": "<div>9</div>",
        },
        Object {
          "id": "0",
          "value": "<div>6</div>",
        },
      ],
      "choicesPosition": "below",
      "correctResponse": Object {
        "0": "0",
        "1": "1",
      },
      "disabled": false,
      "duplicates": true,
      "element": "drag-in-the-blank",
      "feedback": Object {},
      "id": "1",
      "markup": "{{0}} + {{1}} = 15",
      "mode": "evaluate",
      "prompt": "Prompt",
      "promptEnabled": true,
      "rationale": "Rationale",
      "rationaleEnabled": true,
      "shuffle": true,
      "studentInstructionsEnabled": true,
      "teacherInstructions": "Teacher Instructions",
      "teacherInstructionsEnabled": true,
    }
  }
  onChange={[MockFunction]}
  value={
    Object {
      "0": "1",
      "1": "0",
    }
  }
/>
`;

exports[`Main render should render in gather mode 1`] = `
<Main
  classes={
    Object {
      "mainContainer": "Main-mainContainer-1",
    }
  }
  model={
    Object {
      "alternateResponses": Array [
        Array [
          "1",
        ],
        Array [
          "0",
        ],
      ],
      "choices": Array [
        Object {
          "id": "1",
          "value": "<div>9</div>",
        },
        Object {
          "id": "0",
          "value": "<div>6</div>",
        },
      ],
      "choicesPosition": "below",
      "correctResponse": Object {
        "0": "0",
        "1": "1",
      },
      "disabled": false,
      "duplicates": true,
      "element": "drag-in-the-blank",
      "feedback": Object {},
      "id": "1",
      "markup": "{{0}} + {{1}} = 15",
      "mode": "gather",
      "prompt": "Prompt",
      "promptEnabled": true,
      "rationale": "Rationale",
      "rationaleEnabled": true,
      "shuffle": true,
      "studentInstructionsEnabled": true,
      "teacherInstructions": "Teacher Instructions",
      "teacherInstructionsEnabled": true,
    }
  }
  onChange={[MockFunction]}
  value={
    Object {
      "0": "1",
      "1": "0",
    }
  }
/>
`;

exports[`Main render should render in view mode 1`] = `
<Main
  classes={
    Object {
      "mainContainer": "Main-mainContainer-1",
    }
  }
  model={
    Object {
      "alternateResponses": Array [
        Array [
          "1",
        ],
        Array [
          "0",
        ],
      ],
      "choices": Array [
        Object {
          "id": "1",
          "value": "<div>9</div>",
        },
        Object {
          "id": "0",
          "value": "<div>6</div>",
        },
      ],
      "choicesPosition": "below",
      "correctResponse": Object {
        "0": "0",
        "1": "1",
      },
      "disabled": false,
      "duplicates": true,
      "element": "drag-in-the-blank",
      "feedback": Object {},
      "id": "1",
      "markup": "{{0}} + {{1}} = 15",
      "mode": "view",
      "prompt": "Prompt",
      "promptEnabled": true,
      "rationale": "Rationale",
      "rationaleEnabled": true,
      "shuffle": true,
      "studentInstructionsEnabled": true,
      "teacherInstructions": "Teacher Instructions",
      "teacherInstructionsEnabled": true,
    }
  }
  onChange={[MockFunction]}
  value={
    Object {
      "0": "1",
      "1": "0",
    }
  }
/>
`;

exports[`Main render should render without prompt 1`] = `
<Main
  classes={
    Object {
      "mainContainer": "Main-mainContainer-1",
    }
  }
  model={
    Object {
      "alternateResponses": Array [
        Array [
          "1",
        ],
        Array [
          "0",
        ],
      ],
      "choices": Array [
        Object {
          "id": "1",
          "value": "<div>9</div>",
        },
        Object {
          "id": "0",
          "value": "<div>6</div>",
        },
      ],
      "choicesPosition": "below",
      "correctResponse": Object {
        "0": "0",
        "1": "1",
      },
      "disabled": false,
      "duplicates": true,
      "element": "drag-in-the-blank",
      "feedback": Object {},
      "id": "1",
      "markup": "{{0}} + {{1}} = 15",
      "mode": "gather",
      "prompt": null,
      "promptEnabled": true,
      "rationale": "Rationale",
      "rationaleEnabled": true,
      "shuffle": true,
      "studentInstructionsEnabled": true,
      "teacherInstructions": "Teacher Instructions",
      "teacherInstructionsEnabled": true,
    }
  }
  onChange={[MockFunction]}
  value={
    Object {
      "0": "1",
      "1": "0",
    }
  }
/>
`;

exports[`Main render should render without rationale 1`] = `
<Main
  classes={
    Object {
      "mainContainer": "Main-mainContainer-1",
    }
  }
  model={
    Object {
      "alternateResponses": Array [
        Array [
          "1",
        ],
        Array [
          "0",
        ],
      ],
      "choices": Array [
        Object {
          "id": "1",
          "value": "<div>9</div>",
        },
        Object {
          "id": "0",
          "value": "<div>6</div>",
        },
      ],
      "choicesPosition": "below",
      "correctResponse": Object {
        "0": "0",
        "1": "1",
      },
      "disabled": false,
      "duplicates": true,
      "element": "drag-in-the-blank",
      "feedback": Object {},
      "id": "1",
      "markup": "{{0}} + {{1}} = 15",
      "mode": "gather",
      "prompt": "Prompt",
      "promptEnabled": true,
      "rationale": null,
      "rationaleEnabled": true,
      "shuffle": true,
      "studentInstructionsEnabled": true,
      "teacherInstructions": "Teacher Instructions",
      "teacherInstructionsEnabled": true,
    }
  }
  onChange={[MockFunction]}
  value={
    Object {
      "0": "1",
      "1": "0",
    }
  }
/>
`;

exports[`Main render should render without teacher instructions 1`] = `
<Main
  classes={
    Object {
      "mainContainer": "Main-mainContainer-1",
    }
  }
  model={
    Object {
      "alternateResponses": Array [
        Array [
          "1",
        ],
        Array [
          "0",
        ],
      ],
      "choices": Array [
        Object {
          "id": "1",
          "value": "<div>9</div>",
        },
        Object {
          "id": "0",
          "value": "<div>6</div>",
        },
      ],
      "choicesPosition": "below",
      "correctResponse": Object {
        "0": "0",
        "1": "1",
      },
      "disabled": false,
      "duplicates": true,
      "element": "drag-in-the-blank",
      "feedback": Object {},
      "id": "1",
      "markup": "{{0}} + {{1}} = 15",
      "mode": "gather",
      "prompt": "Prompt",
      "promptEnabled": true,
      "rationale": "Rationale",
      "rationaleEnabled": true,
      "shuffle": true,
      "studentInstructionsEnabled": true,
      "teacherInstructions": null,
      "teacherInstructionsEnabled": true,
    }
  }
  onChange={[MockFunction]}
  value={
    Object {
      "0": "1",
      "1": "0",
    }
  }
/>
`;
